intro-HPC/ch_checkpointing.tex

```diff
@@ -1,4 +1,177 @@
+An alternative is to copy the interesting files to the shared storage at the end of the job script.
+
+
+\subsection{Running on shared storage (\texttt{--shared})}
+
+If the job needs to be run on the shared storage, \lstinline|--shared| should be specified.
+You should enable this option by default, because it makes the execution of the
+underlying \lstinline|csub| script more robust: it doesn't have to copy from/to
+the local storage on the workernode. When enabled, the job will be run in a
+subdirectory of \lstinline|$VSC_SCRATCH/chkpt|. All files produced by the job will
+be in \lstinline|$VSC_SCRATCH/chkpt/<jobid>/| while the job is running.
+
+Note that if files in the directory where the job script is located are required to
+run the job, you should also use \lstinline|--pre|.
+
+\subsection{Pro/epilogue mimicing (\texttt{--no\_mimic\_pro\_epi})}
+
+The option \lstinline|--no_mimic_pro_epi| disables the workaround currently required to resolve
+a permissions problem when using actual Torque prologue/epilogue scripts. Don't use
+this option unless you really know what you're doing.
+
+\subsection{Job wall time (\texttt{--job\_time}, \texttt{--chkpt\_time})}
+
+To specify the requested wall time per subjob, use the \lstinline|--job-time| parameter.
+The default setting is 10 hours per (sub)job. Lowering this will result in more
+frequent checkpointing, and thus more (sub)jobs.
+
+To specify the time that is reserved for checkpointing the job, use \lstinline|--chkpt_time|.
+By default, this is set to 15 minutes which should be enough for most applications/jobs.
+\emph{Don't change this unless you really need to}.
+
+The total requested wall time per subjob is the sum of both \lstinline|job_time| and \lstinline|chkpt_time|.
+This should be taken into account when submitting to a specific job queue
+(e.g. \lstinline|debug| which only support jobs of up to 1 hour).
+
+If you would like to time how long the job executes, just prepend the main command in your job script with
+the time command \lstinline|time|, e.g.:
+
+\begin{prompt}
+time main_command
+\end{prompt}
+
+The \lstinline|real| time will not make sense, as it will also include the time
+passed between two checkpointed subjobs. However, the \lstinline|user| time should give a good
+indication of the actual time it took to run your command, even if multiple checkpoints were performed.
+
+\subsection{Cleanup checkpoints (\texttt{--cleanup\_after\_restart})}
+
+Specifying this option will make the wrapper script remove the checkpoint files after
+a successful job restart. This may be desirable in cause you're short on storage space.
+
+Note that we don't recommend setting this option, because this way you won't be able
+to resume from the last checkpoint when something goes wrong. It may also prevent
+the wrapper script from reattempting to resubmit a new job in case an infrequent known
+failure occurs. So, don't set this unless you really need to.
```

Comment:
All these "useful-but-don't-use-this" options should be moved into a separate "Additional options" section, rather than mixing them with the more important options like `-s` and `--job-time`